services/notifications/controller/controller.go

```diff
@@ -27,6 +28,44 @@ func SetupController(route *mux.Route) {
 	router.HandleFunc("/order/{id}/", GetNotificationOrder).Methods("GET")
 }
 
+/*
+	Returns all topics a user is subscribed to
+*/
+func GetAllSubscriptions(w http.ResponseWriter, r *http.Request) {
+	id := r.Header.Get("HackIllinois-Identity")
+	topics, err := service.GetSubscriptions(id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not retrieve notifications."))
+		return
+	}
+
+	topic_list := models.TopicList{
+		Topics: topics,
+	}
+
+	json.NewEncoder(w).Encode(topic_list)
+}
+
+/*
+	Returns devices registered of a specific user
+*/
+func GetRegisteredDevices(w http.ResponseWriter, r *http.Request) {
+	id := r.Header.Get("HackIllinois-Identity")
+	devices, err := service.GetUserDevices(id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not retrieve notifications."))
```

Comment:
change this too

Reply:
Hi Heesoo, 

After revisiting, I think you're right. My interpretation of the PR was a bit different. Based on the work done so far, I'd like to merge these into master and hopefully open a new PR to address #247 more properly soon. Additionally, I think this can count as my "coding challenge" for the API team right? Please get back to me when you get a chance : )